examples/run_nlp_trainer.py

```diff
@@ -0,0 +1,439 @@
+    return trainer.state.current_metrics
+
+
+def get_args() -> argparse.Namespace:
+    """Get NLPTrainerHparams arguments from CLI."""
+    parser = hp.get_argparse(NLPTrainerHparams)
+    parser.add_argument('--training_scheme',
+                        help='training scheme used (one of "pretrain", "finetune", or "all")',
+                        choices=['pretrain', 'finetune', 'all'],
+                        required=True)
+
+    args = parser.parse_args()
+
+    return args
+
+
+def validate_args(args: argparse.Namespace, hp: NLPTrainerHparams) -> None:
+    """Validate CLI args as well as finetune-specific parameters."""
+    if not args.file:
+        raise ValueError(
+            'must specify a top level file to store shared configs for the entrypoint. see composer/yamls/models/glue/glue_example.yaml for an example.'
+        )
+
+    if args.training_scheme == 'finetune' and (not hp.finetune_hparams) or (hp.finetune_hparams and
+                                                                            not hp.finetune_hparams.load_path):
+        raise ValueError('load_path to checkpoint folder must be specified if finetuning a model')
+
+    elif args.training_scheme == 'all' and hp.finetune_hparams is None:
+        warnings.warn('No shared finetune_hparams specified. All finetune tasks will use their default configurations.')
+        hp.finetune_hparams = GLUETrainerHparams()
+
+    elif args.training_scheme == 'pretrain' and hp.finetune_hparams:
+        warnings.warn('finetune_hparams specified. These values will be ignored during pretraining.')
+
+    elif args.training_scheme == 'all' and hp.finetune_hparams and hp.finetune_hparams.load_path:
+        warnings.warn('load_path specified in finetune_hparams. This value will be overriden during finetuning.')
+
+
+def get_finetune_hparams(args: argparse.Namespace) -> Union[GLUETrainerHparams, None]:
+    """Extract finetune-specific hparams from the provided file and add entrypoint specific args to it."""
+    hp = NLPTrainerHparams.create(cli_args=False, f=args.file)
+    validate_args(args, hp)
+
+    if args.training_scheme != 'pretrain':
+        hparams = hp.finetune_hparams
+        args.save_locally = True
+        if hparams and hparams.loggers:
+            for l in hparams.loggers:
+                if isinstance(l, ObjectStoreLogger):
+                    args.save_locally = False
+                    args.bucket = l.object_store.bucket
+
+        if args.training_scheme == 'all':
+            hparams.save_folder = hp.pretrain_hparams.save_folder
+
+    return hparams
+
+
+def run_pretrainer(args) -> None:
+    """Logic for handling a pretraining job spawn based on storage and training settings."""
+    root_dir = os.path.join(os.path.dirname(composer.__file__), '..')
+    training_script = os.path.join(root_dir, 'examples/run_composer_trainer.py')
+
+    # manually copy pretrain_hparams to temporary file (workaround until CO-766 resolved)
```

Comment:
This is the issue I was referring to: https://mosaicml.atlassian.net/browse/CO-766. Originally the plan was to extract just the pretrain_hparams and then dump the `.to_yaml()` to a file (which is much simpler), but since YAHP has this bug I can't use that trick.

Reply:
I'll file a JIRA to back fix this after the YAHP fixes are released!